src/luarocks/fetch/git.lua

```diff
@@ -112,7 +112,8 @@ function git.get_sources(rockspec, extract, dest_dir, depth)
    local ok, err = fs.change_dir(store_dir)
    if not ok then return nil, err end
 
-   local command = {fs.Q(git_cmd), "clone", depth or "--depth=1", rockspec.source.url, module}
+   local https_source_url = "https://" .. rockspec.source.url
```

Comment:
Just putting https at the front isn't going to be correct. e.g. the url in the rockspec might already have `https` in the front.
Alternatively it could be `ssh://`, or numerous other options.